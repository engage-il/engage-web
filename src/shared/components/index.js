// @flow
export { Link } from './Link'
export { CopyLink } from './CopyLink'
export { Button } from './Button'
export { NotificationView } from './NotificationView'
