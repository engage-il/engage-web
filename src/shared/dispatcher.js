// @flow
import Events from 'ampersand-events'

const dispatcher = Events.createEmitter()

type Event = 'menu-changed'
type EventHandler = (args: any) => void

export function dispatch (event: Event, args: any) {
  dispatcher.trigger(event, args)
}

export function on (event: Event, handler: EventHandler) {
  dispatcher.on(event, handler)
}

export function off (event: Event, handler: EventHandler) {
  dispatcher.off(event, handler)
}
