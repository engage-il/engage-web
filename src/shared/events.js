// @flow
import EventEmitter from 'eventemitter3'

class Events extends EventEmitter {
  setAuthHeader = '@@events/set-auth-header'
  setEnvironment = '@@events/set-environment'
  showNotification = '@@events/show-notification'
}

export const events = new Events()
