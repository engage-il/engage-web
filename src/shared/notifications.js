// @flow
export type Notification = {
  key: string,
  message: string
}

type NotificationListener = (Notification) => (void | Promise<void>)

let _listener: ?NotificationListener = null

export const notifications = {
  on (listener: NotificationListener) {
    _listener = listener
  },
  off () {
    _listener = null
  },
  add (notification: Notification) {
    _listener && _listener(notification)
  }
}
