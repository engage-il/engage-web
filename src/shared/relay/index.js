// @flow
export { currentEnvironment } from './currentEnvironment'
export { unwrap } from './unwrap'
export { withLoadMoreArgs } from './withLoadMoreArgs'
export { cacheResolvers } from './cacheResolvers'
export { createPaginationCacheResolver } from './createPaginationCacheResolver'
