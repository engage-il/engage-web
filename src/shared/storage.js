/* globals Storage, localStorage, sessionStorage */
// @flow

function namespaceKey (key: string): string {
  return `@@legislated/${key}`
}

class Store<K: string> {
  storage: Storage

  constructor (storage: Storage) {
    this.storage = storage
  }

  set (key: K, value: ?string) {
    const actualKey = namespaceKey(key)

    if (value) {
      this.storage.setItem(actualKey, value)
    } else {
      this.storage.removeItem(actualKey)
    }
  }

  get (key: K): ?string {
    const actualKey = namespaceKey(key)
    return this.storage.getItem(actualKey)
  }
}

type LocalStoreKey
  = 'intro-visited'
  | 'intro-cleared'

type SessionStoreKey
  = 'last-search-count'
  | 'admin-header'

export const local: Store<LocalStoreKey> = new Store(localStorage)
export const session: Store<SessionStoreKey> = new Store(sessionStorage)
