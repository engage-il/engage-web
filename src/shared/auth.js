// @flow
import { session } from 'shared/storage'
import { events } from 'shared/events'
import { Base64 } from 'js-base64'

export const auth = {
  get isSignedIn (): boolean {
    return !!session.get('admin-header')
  },
  signIn (username: string, password: string) {
    const authValue = Base64.encode(`${username}:${password}Â£`)
    const authHeader = `Basic ${authValue}`
    session.set('admin-header', authHeader)
    events.emit(events.setAuthHeader, authHeader)
  },
  signOut () {
    session.set('admin-header', null)
    events.emit(events.setAuthHeader, null)
  }
}
