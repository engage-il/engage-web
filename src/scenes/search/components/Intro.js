// @flow
import React, { Component } from 'react'
import FontAwesome from 'react-fontawesome'
import { css } from 'glamor'
import type { Rule } from 'glamor'
import { BillAnimation } from './BillAnimation'
import { Link } from 'shared/components'
import { local } from 'shared/storage'
import { stylesheet, mixins } from 'shared/styles'

export class Intro extends Component {
  props: {
    styles?: Rule
  }

  state = {
    isAccepted: false
  }

  // events
  didClickAccept = () => {
    this.setState({ isAccepted: true }, () => {
      local.set('intro-cleared', 'true')
    })
  }

  // lifecycle
  componentWillMount () {
    local.set('intro-visited', 'true')
  }

  render () {
    return local.get('intro-cleared') ? null : <BillAnimation>
      {this.renderContent()}
    </BillAnimation>
  }

  renderContent (): ?React$Element<*> {
    if (this.state.isAccepted) {
      return null
    }

    const { styles } = this.props

    return <div {...css(rules.intro, styles)}>
      <h1>Hey, welcome to Legislated!</h1>
      <p>
        Never heard of a witness slip? As a resident of Illinois, it's a tool
        that allows you to voice your opinion on the laws your representatives
        write. Search for a bill you care about and let the legislature know
        how you feel. <strong>Want to learn more?</strong>
        <span {...rules.actionLinks}>
          <Link to='/faq'>Visit our FAQs</Link>
          <Link styles={rules.accept} onClick={this.didClickAccept}>
            <FontAwesome name='check' />
            <span>Got it.</span>
          </Link>
        </span>
      </p>
    </div>
  }
}

const rules = stylesheet({
  intro: {
    fontSize: 18,
    '> h1': {
      marginBottom: 15
    }
  },
  actionLinks: {
    marginLeft: 10,
    ...mixins.mobile({
      display: 'block',
      marginTop: 10,
      marginLeft: 0
    })
  },
  accept: {
    ...mixins.borders.low(['left']),
    marginLeft: 10,
    paddingLeft: 10,
    '> *:last-child': {
      marginLeft: 5
    }
  }
})
