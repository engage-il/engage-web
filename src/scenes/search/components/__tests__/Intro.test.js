/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { Intro } from '../Intro'
import { local } from 'shared/storage'

let subject
let props

function loadSubject () {
  subject = shallow(<Intro {...props} />)
}

function test () {
  expect(subject).toMatchSnapshot()
}

const element = {
  accept: () => subject.find('Link').at(1)
}

beforeEach(() => {
  props = {}
  subject = null
})

describe('#render', () => {
  it('renders correctly', () => {
    loadSubject()
    test()
  })
  it('when it is cleared', () => {
    local.set('intro-cleared', 'true')
    loadSubject()
    test()
  })
  it('when the user clicks accept', () => {
    loadSubject()
    element.accept().simulate('click')
    test()
  })
})
