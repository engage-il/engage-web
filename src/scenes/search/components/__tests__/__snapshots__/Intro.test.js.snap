// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render renders correctly 1`] = `
<BillAnimation>
  <div
    data-css-k92j1l=""
  >
    <h1>
      Hey, welcome to Legislated!
    </h1>
    <p>
      Never heard of a witness slip? As a resident of Illinois, it's a tool that allows you to voice your opinion on the laws your representatives write. Search for a bill you care about and let the legislature know how you feel. 
      <strong>
        Want to learn more?
      </strong>
      <span
        data-css-1flwhvu=""
      >
        <Link
          to="/faq"
        >
          Visit our FAQs
        </Link>
        <Link
          onClick={[Function]}
          styles={
            Object {
              "data-css-16kle8p": "",
            }
          }
        >
          <FontAwesome
            name="check"
          />
          <span>
            Got it.
          </span>
        </Link>
      </span>
    </p>
  </div>
</BillAnimation>
`;

exports[`#render when it is cleared 1`] = `null`;

exports[`#render when the user clicks accept 1`] = `<BillAnimation />`;
